src/main/java/frc/robot/subsystems/vision/VisionIOPhotonLib.java

```diff
@@ -0,0 +1,197 @@
+package frc.robot.subsystems.vision;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Optional;
+
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+import org.photonvision.targeting.PhotonPipelineResult;
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.networktables.NetworkTableEvent;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import frc.robot.Constants;
+import frc.robot.subsystems.drive.Drive;
+
+public class VisionIOPhotonLib implements VisionIO
+{
+    private final PhotonCamera  _camera;
+    private double              _captureTimesStamp = 0.0;
+    private Pose2d              _pose              = new Pose2d();
+    private boolean             _hasPose           = false;
+    private double[]            _distances         = new double[] {};
+    private int[]               _targetIds         = new int[] {};
+    private double[]            _targetYaws        = new double[] {};
+    private int                 _numProcessedTargets;
+    private AprilTagFieldLayout fieldLayout;
+    private PhotonPoseEstimator _poseEstimator;
```

Comment:
The field layout and pose estimator can both be `final`. Also, `fieldLayout` doesn't follow the naming convention we use for member variables:

```suggestion
    private final PhotonCamera        _camera;
    private final AprilTagFieldLayout _fieldLayout;
    private final PhotonPoseEstimator _poseEstimator;
    private double                    _captureTimesStamp = 0.0;
    private Pose2d                    _pose              = new Pose2d();
    private boolean                   _hasPose           = false;
    private double[]                  _distances         = new double[] {};
    private int[]                     _targetIds         = new int[] {};
    private double[]                  _targetYaws        = new double[] {};
    private int                       _numProcessedTargets;
```